habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -360,22 +375,30 @@ def _add_objs(self, ep_info: Config, should_add_objects: bool) -> None:
         for aoi_handle in ao_mgr.get_object_handles():
             self.art_objs.append(ao_mgr.get_object_by_handle(aoi_handle))
 
-    def _create_obj_viz(self, ep_info: Config) -> None:
-        if self._debug_render:
-            for marker_name, m in self._markers.items():
-                m_T = m.get_current_transform()
-                self.viz_ids[marker_name] = self.visualize_position(
-                    m_T.translation, self.viz_ids[marker_name]
-                )
+    def _create_obj_viz(self, ep_info):
+        for marker_name, m in self._markers.items():
+            m_T = m.get_current_transform()
+            self.viz_ids[marker_name] = self.visualize_position(
+                m_T.translation, self.viz_ids[marker_name]
+            )
 
-        # TODO: refactor this
-        # target_name_pos = [
-        #     (ep_info["static_objs"][idx][0], self.scene_objs[idx], pos)
-        #     for idx, pos in self._get_target_trans()
-        # ]
-        # self.viz_obj_ids = place_viz_objs(
-        #     target_name_pos, self, self.viz_obj_ids
-        # )
+        rom = self.get_rigid_object_manager()
+        obj_attr_mgr = self.get_object_template_manager()
+        for target_handle, transform in self._targets.items():
+            new_target_handle = (
+                target_handle.split("_:")[0] + ".object_config.json"
+            )
+            matching_templates = (
+                obj_attr_mgr.get_templates_by_handle_substring(
+                    new_target_handle
+                )
+            )
+            ro = rom.add_object_by_template_handle(
+                list(matching_templates.keys())[0]
```

Comment:
```suggestion
                next(iter(matching_templates.keys()))
```
^ Has O(n) complexity